From fc50087e6b58c9208e84b49dc87bed919c44c303 Mon Sep 17 00:00:00 2001
From: MisterZtr <andey2004586@gmail.com>
Date: Mon Dec 9 06:27:27 2024 +0300
Subject: Settings: Simplify getAvailabilityStatus() in ForceLteCaPreferenceController

diff --git a/src/com/android/settings/network/telephony/ForceLteCaPreferenceController.java b/src/com/android/settings/network/telephony/ForceLteCaPreferenceController.java
index 57535585c4b..acedb8a84a3 100644
--- a/src/com/android/settings/network/telephony/ForceLteCaPreferenceController.java
+++ b/src/com/android/settings/network/telephony/ForceLteCaPreferenceController.java
@@ -58,16 +58,12 @@ public class ForceLteCaPreferenceController extends TelephonyTogglePreferenceCon
     }
     @Override
     public int getAvailabilityStatus(int subId) {
-
         if (!mContext.getResources().getBoolean(R.bool.config_showForceLteCaToggle))
             return UNSUPPORTED_ON_DEVICE;
 
         init(subId);
-        if (SubscriptionManager.isValidSubscriptionId(mSubId)) {
-            return AVAILABLE;
-        }
-
-        return CONDITIONALLY_UNAVAILABLE;
+        return (SubscriptionManager.isValidSubscriptionId(mSubId))
+            ? AVAILABLE : CONDITIONALLY_UNAVAILABLE;
     }
     @Override
     public void displayPreference(PreferenceScreen screen) {
