From 07ccd9160d19119c904d8bf2f80c683d7b880214 Mon Sep 17 00:00:00 2001
From: MisterZtr <andey2004586@gmail.com>
Date: Tue Dec 3 06:08:26 2024 +0300
Subject: [PATCH] build: Prefer Aperture over Camera2

diff --git a/target/product/handheld_product.mk b/target/product/handheld_product.mk
index b2ce4c5e19..3864555d96 100644
--- a/target/product/handheld_product.mk
+++ b/target/product/handheld_product.mk
@@ -35,7 +35,7 @@ PRODUCT_PACKAGES += \
 
 ifeq ($(LINEAGE_BUILD),)
 PRODUCT_PACKAGES += \
-    Camera2 \
+    Aperture \
     LatinIME
 endif

