From bdcba7bb7e5a6a076ec0b1ddeaac6dcad0070e86 Mon Sep 17 00:00:00 2001
From: MisterZtr <andey2004586@gmail.com>
Date: Sun Jul 9 19:44:49 2023 +0300
Subject: [PATCH] Dont-handle-apns.patch

diff --git a/target/product/aosp_product.mk b/target/product/aosp_product.mk
index 67ab7a7433..5dae7dbc84 100644
--- a/target/product/aosp_product.mk
+++ b/target/product/aosp_product.mk
@@ -36,10 +36,3 @@ PRODUCT_PACKAGES += \
     PhotoTable \
     preinstalled-packages-platform-aosp-product.xml \
     ThemePicker \
-
-# Telephony:
-#   Provide a APN configuration to GSI product
-ifeq ($(LINEAGE_BUILD),)
-PRODUCT_COPY_FILES += \
-    device/sample/etc/apns-full-conf.xml:$(TARGET_COPY_OUT_PRODUCT)/etc/apns-conf.xml
-endif
diff --git a/target/product/full_base_telephony.mk b/target/product/full_base_telephony.mk
index e4f5a162bb..5470ad2be5 100644
--- a/target/product/full_base_telephony.mk
+++ b/target/product/full_base_telephony.mk
@@ -26,10 +26,5 @@ PRODUCT_VENDOR_PROPERTIES := \
 PRODUCT_COPY_FILES := \
     frameworks/native/data/etc/handheld_core_hardware.xml:$(TARGET_COPY_OUT_VENDOR)/etc/permissions/handheld_core_hardware.xml
 
-ifeq ($(LINEAGE_BUILD),)
-PRODUCT_COPY_FILES += \
-    device/sample/etc/apns-full-conf.xml:system/etc/apns-conf.xml
-endif
-
 $(call inherit-product, $(SRC_TARGET_DIR)/product/aosp_base.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/telephony.mk)
