From dd702d06819da13b220f657509200733d8f3b1e1 Mon Sep 17 00:00:00 2001
From: MisterZtr <andey2004586@gmail.com>
Date: Sun Dec 8 12:29:11 2024 +0300
Subject: lineage: use own config

diff --git a/lineage.mk b/lineage.mk
index 172bb01..5f01894 100644
--- a/lineage.mk
+++ b/lineage.mk
@@ -1,3 +1,20 @@
+$(call inherit-product, vendor/lineage/config/common_full_tablet.mk)
 $(call inherit-product, vendor/lineage/config/common_full_phone.mk)
+$(call inherit-product, vendor/lineage/config/BoardConfigSoong.mk)
+$(call inherit-product, vendor/lineage/config/BoardConfigLineage.mk)
+$(call inherit-product, device/lineage/sepolicy/common/sepolicy.mk)
 -include vendor/lineage/build/core/config.mk
--include vendor/lineage/build/core/apicheck.mk
+TARGET_NO_KERNEL_OVERRIDE := true
+TARGET_NO_KERNEL_IMAGE := true
+SELINUX_IGNORE_NEVERALLOWS := true
+TARGET_USES_PREBUILT_VENDOR_SEPOLICY := true
+TARGET_HAS_FUSEBLK_SEPOLICY_ON_VENDOR := true
+
+# Bootanimation
+TARGET_BOOT_ANIMATION_RES := 1080
+
+# Face unlock
+TARGET_FACE_UNLOCK_SUPPORTED := true
+
+# APN
+PRODUCT_PACKAGES += apns-conf.xml
