From ca7d643c811a3dbd5f09b0b657940914a9f770a3 Mon Sep 17 00:00:00 2001
From: MisterZtr <andey2004586@gmail.com>
Date: Sun Dec 8 12:35:20 2024 +0300
Subject: base: providing a choice of additional vndk to build image with ext4

diff --git a/base.mk b/base.mk
index b491efe..9d21161 100644
--- a/base.mk
+++ b/base.mk
@@ -260,5 +260,3 @@ PRODUCT_PACKAGES += \
 $(call inherit-product, $(SRC_TARGET_DIR)/product/window_extensions.mk)
 PRODUCT_PRODUCT_PROPERTIES += \
     persist.settings.large_screen_opt.enabled=true
-
-PRODUCT_EXTRA_VNDK_VERSIONS += 28 29
