From 19c590062250f7e5dcb59daac47124cffdd3fb3c Mon Sep 17 00:00:00 2001
From: MisterZtr <andey2004586@gmail.com>
Date: Fri Dec 20 21:44:01 2024 +0300
Subject: props: Set ro.control_privapp_permissions

diff --git a/system.prop b/system.prop
index d8fba53..09ad957 100644
--- a/system.prop
+++ b/system.prop
@@ -52,3 +52,6 @@ vendor.usb.use_ffs_mtp=1
 
 # Force LTE_CA
 persist.sys.radio.force_lte_ca=true
+
+# PRIV-APP
+ro.control_privapp_permissions=log
