From Ieb0de91723dae11310977a0dd74bccadb73c4e2c Mon Sep 17 00:00:00 2001
From: MisterZtr <andey2004586@gmail.com>
Date: Sun Dec 8 11:16:34 2024 +0300
Subject: bluetooth: Prefer os::Get over Get

diff --git a/system/gd/hci/controller.cc b/system/gd/hci/controller.cc
index d65f766332..b58206d8c6 100644
--- a/system/gd/hci/controller.cc
+++ b/system/gd/hci/controller.cc
@@ -600,7 +600,7 @@ struct Controller::impl {
     vendor_capabilities_.is_supported_ = 0x01;
 
     int vendor_cap_max = 0xffff;
-    std::string vendor_cap_max_prop = GetSystemProperty("persist.sys.bt.max_vendor_cap").value_or("");
+    std::string vendor_cap_max_prop = os::GetSystemProperty("persist.sys.bt.max_vendor_cap").value_or("");
     if (vendor_cap_max_prop != "") {
         vendor_cap_max = std::stoi(vendor_cap_max_prop);
     }
