From 51eb27fd3c5c69876cf4032242b779db43c6b86d (HEAD)
From: MisterZtr <andey2004586@gmail.com>
Date: Mon Apr 7 09:59:30 2025 +0300
Subject: [PATCH] init: fix oppo tfa98

diff --git a/rootdir/init.rc b/rootdir/init.rc
index 0cc129707..09301cdda 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -1257,6 +1257,10 @@ on boot
     # Define default initial receive window size in segments.
     setprop net.tcp_def_init_rwnd 60
 
+    # Fix oppo tfa98
+    write /proc/tfa98xx/oppo_tfa98x_fw_update 1
+    chown system system /proc/tfa98xx/oppo_tfa98x_fw_update
+
     # Start standard binderized HAL daemons
     class_start hal
 
